tripy/nvtripy/frontend/tensor.py

```diff
@@ -224,11 +225,21 @@ def eval(self) -> runtime.MemRefValue:
         assert len(data) == 1, "Expects only one output from mlir_tensorrt.compiler executor"
         data = data[0]
 
+        # If we are tracing for compilation, we record the change we make so that we can revert it before producing
+        # the final executable. However, even so, evaluating while tracing could introduce errors
+        # (e.g., if the result of evaluation is used to create a constant that is in the graph),
+        # so we still give warnings in that case.
+        #
+        # In principle, we could avoid inserting the storage op altogether, but that could slow down other
+        # evaluations that occur during the trace.
+        if REVERT_GRAPH_AFTER_COMPILING is not None:
+            REVERT_GRAPH_AFTER_COMPILING.append((self.trace_tensor, self.trace_tensor.producer))
```

Comment:
This might be a bit of a clumsy approach to keeping around the old producers (required a global var from the compiler). I wasn't sure where else the data structure could reside, since evaluation is handled as a method on tensors. It's a tricky issue because the tensor evaluated could be anywhere in the middle of the graph and the interface to `compile` is an opaque `Callable`.

Reply:
What about storing it on the trace tensors themselves? `original_producer` or something like that?

Reply:
I guess in that case we would need to do a DFS from the output and swap the producer before stepping further

Reply:
Yeah searching the graph might be necessary if there isn't anywhere else we could record the tensors that have been evaluated.